intake/catalog/tests/test_auth_integration.py

```diff
@@ -0,0 +1,73 @@
+import os
+import os.path
+import shutil
+import tempfile
+import time
+
+import pytest
+
+from .util import assert_items_equal
+from intake.catalog import Catalog
+
+from intake.auth.secret import SecretClientAuth
+
+TMP_DIR = tempfile.mkdtemp()
+CONF_DIR = os.path.join(TMP_DIR, 'conf')
+os.mkdir(CONF_DIR)
+
+TEST_CATALOG_PATH = [TMP_DIR]
+YAML_FILENAME = 'intake_test_catalog.yml'
+
+
+# Create server configuration using shared-secret Auth
+TEST_SERVER_CONF = os.path.join(CONF_DIR, 'config.yaml')
+conf = '''
+auth:
+  class: intake.auth.secret.SecretAuth
+  kwargs:
+    secret: test_secret
+'''
+with open(TEST_SERVER_CONF, 'w') as f:
+    f.write(conf)
+
+
+def teardown_module(module):
+    shutil.rmtree(TMP_DIR)
+
+
+@pytest.fixture
+def intake_server_with_auth(intake_server):
+    fullname = os.path.join(TMP_DIR, YAML_FILENAME)
+
+    with open(fullname, 'w') as f:
+        f.write('''
```

Comment:
Maybe we should include MemoryFS to avoid having to write all these temporary files... or be able to directly pass a string to Catalog(), without assuming it's a path to open.